### ISSUE TITLE
Test failure  JIT\\opt\\Tailcall\\TailcallVerifyWithPrefix\\TailcallVerifyWithPrefix.cmd

### ISSUE THREAD

**Issue by v-wenyuxu** [dotnet/runtime#79517]
Failed in Run: [runtime-coreclr outerloop 20221211.2](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_build/results?buildId=108789)
 
 **Failed tests:**
 ```
coreclr windows arm Checked no_tiered_compilation @ Windows.10.Arm64v8.Open

- JIT\\opt\\Tailcall\\TailcallVerifyWithPrefix\\TailcallVerifyWithPrefix.cmd

coreclr windows x64 Checked no_tiered_compilation @ Windows.10.Amd64.Open

- JIT\\opt\\Tailcall\\TailcallVerifyWithPrefix\\TailcallVerifyWithPrefix.cmd 
```
 
 **Error message:**
 ```
 
Return code:      1
Raw output file:      D:\h\w\AA0E0926\w\B52D098D\uploads\Reports\JIT.opt\Tailcall\TailcallVerifyWithPrefix\TailcallVerifyWithPrefix.output.txt
Raw output:
BEGIN EXECUTION
 "D:\h\w\AA0E0926\p\corerun.exe" -p "System.Reflection.Metadata.MetadataUpdater.IsSupported=false"  TailcallVerifyWithPrefix.dll 
Executing Condition3.Test1 - Caller: Arguments: None - ReturnType: Int32; Callee: Arguments: None - ReturnType: Int32
Execution finished - Test PASSED
Executing Condition3.Test2 - Caller: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: string
Execution finished - Test PASSED
Executing Condition4.Test1 - Caller: Arguments: None - ReturnType: 3 byte struct; Callee: Arguments: None - ReturnType: 3 byte struct
Execution finished - Test PASSED
Executing Condition4.Test2 - Caller: Arguments: None - ReturnType: 3 byte struct; Callee: Arguments: None - ReturnType: 3 byte struct [Verifying the field values in the return type struct]
Execution finished - Test PASSED
Executing Condition5.Test1 - There is a 'conv.i2' IL instruction between the tail. call to the Callee and the ret IL instruction which is not allowed. Expecting to catch an InvalidProgramException internally if tail. prefix is present.
Execution finished - Test PASSED
Executing Condition5.Test2 - Caller: Arguments: None - ReturnType: byte; Callee: Arguments: None - ReturnType: byte
Execution finished - Test PASSED
Executing Condition5.Test3 - Caller: Arguments: None - ReturnType: Int32; Callee: Arguments: None - ReturnType: Int32
Execution finished - Test PASSED
Executing Condition5.Test4 - Caller: Arguments: None - ReturnType: UInt32; Callee: Arguments: None - ReturnType: Int32
Execution finished - Test PASSED
Executing Condition5.Test5 - Caller: Arguments: None - ReturnType: Int32; Callee: Arguments: None - ReturnType: UInt32
Execution finished - Test PASSED
Executing Condition5.Test6 - Caller: Arguments: None - ReturnType: Int64; Callee: Arguments: None - ReturnType: UInt64
Execution finished - Test PASSED
Executing Condition5.Test7 - Caller: Arguments: None - ReturnType: UInt64; Callee: Arguments: None - ReturnType: Int64
Execution finished - Test PASSED
Executing Condition5.Test8 - Caller: Arguments: None - ReturnType: UInt64; Callee: Arguments: None - ReturnType: UInt32 - Expecting a InvalidProgramException
Execution finished - Test PASSED
Executing Condition5.Test9 - Caller: Arguments: None - ReturnType: UInt32; Callee: Arguments: None - ReturnType: UInt64 - Expecting a InvalidProgramException
Execution finished - Test PASSED
Executing Condition9.Test1 - Caller: Arguments: None - ReturnType: void; Callee(Interface Member): Arguments: None - ReturnType: void
Execution finished - Test PASSED
Executing Condition10.Test1 - Caller<string>: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: void
Execution finished - Test PASSED
Executing Condition10.Test2 - Caller: Arguments: None - ReturnType: void; Callee<string>: Arguments: None - ReturnType: void
Execution finished - Test PASSED
Executing Condition10.Test3 - class Foo1<string, int>: Caller: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: void
                              class Foo1<string, int>: Caller: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: void
                              class Foo1<string, short>: Caller: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: void
                              class Foo1<string, ValueType3Bytes>: Caller: Arguments: None - ReturnType: void; Callee: Arguments: None - ReturnType: void
Execution finished - Test PASSED
Executing Condition10.Test4 - class Foo2<string, int>: Caller: Arguments: string, int - ReturnType: void; Callee: Arguments: string - ReturnType: void
                              class Foo2<string, int>: Caller: Arguments: string, int - ReturnType: void; Callee: Arguments: string - R
 ```
 
 **Stack trace:**
 ```
    at JIT_opt._Tailcall_TailcallVerifyWithPrefix_TailcallVerifyWithPrefix_._Tailcall_TailcallVerifyWithPrefix_TailcallVerifyWithPrefix_cmd()
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodInvoker.Invoke(Object obj, IntPtr* args, BindingFlags invokeAttr)
 ```

```json
{
  "ErrorMessage": "FAILED: Callee1Recursive was not tailcall",
  "BuildRetry": false
}
```

### Report

|Build|Definition|Test|Pull Request|
|---|---|---|---|
|[109365](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109365)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109365&view=ms.vss-test-web.build-test-results-tab&runId=2336274&resultId=113321)|dotnet/runtime#77597|
|[109273](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109273)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109273&view=ms.vss-test-web.build-test-results-tab&runId=2335056&resultId=113321)|dotnet/runtime#79538|
|[109118](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109118)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109118&view=ms.vss-test-web.build-test-results-tab&runId=2330930&resultId=113321)|dotnet/runtime#77480|
|[109193](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109193)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109193&view=ms.vss-test-web.build-test-results-tab&runId=2330608&resultId=113321)|dotnet/runtime#79346|
|[109167](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109167)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109167&view=ms.vss-test-web.build-test-results-tab&runId=2329480&resultId=113321)|dotnet/runtime#79530|
|[109094](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109094)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109094&view=ms.vss-test-web.build-test-results-tab&runId=2328146&resultId=113321)|dotnet/runtime#79523|
|[109070](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109070)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109070&view=ms.vss-test-web.build-test-results-tab&runId=2327658&resultId=113321)|dotnet/runtime#78888|
|[109061](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109061)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=109061&view=ms.vss-test-web.build-test-results-tab&runId=2326380&resultId=200807)||
|[108984](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108984)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108984&view=ms.vss-test-web.build-test-results-tab&runId=2325746&resultId=113321)|dotnet/runtime#79519|
|[108931](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108931)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108931&view=ms.vss-test-web.build-test-results-tab&runId=2325380&resultId=319257)||
|[108980](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108980)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108980&view=ms.vss-test-web.build-test-results-tab&runId=2324608&resultId=113321)|dotnet/runtime#79489|
|[108895](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108895)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108895&view=ms.vss-test-web.build-test-results-tab&runId=2323684&resultId=120873)||
|[108858](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108858)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108858&view=ms.vss-test-web.build-test-results-tab&runId=2322196&resultId=113321)|dotnet/runtime#78280|
|[108842](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108842)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108842&view=ms.vss-test-web.build-test-results-tab&runId=2321730&resultId=200809)|dotnet/runtime#79346|
|[108827](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108827)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108827&view=ms.vss-test-web.build-test-results-tab&runId=2320496&resultId=114644)|dotnet/runtime#76642|
|[108817](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108817)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108817&view=ms.vss-test-web.build-test-results-tab&runId=2319416&resultId=200809)||
|[108807](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108807)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108807&view=ms.vss-test-web.build-test-results-tab&runId=2318964&resultId=113321)|dotnet/runtime#75054|
|[108784](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108784)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108784&view=ms.vss-test-web.build-test-results-tab&runId=2318388&resultId=113321)|dotnet/runtime#79501|
|[108745](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108745)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108745&view=ms.vss-test-web.build-test-results-tab&runId=2318328&resultId=113321)|dotnet/runtime#79504|
|[108739](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108739)|dotnet/runtime|[JIT.1.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108739&view=ms.vss-test-web.build-test-results-tab&runId=2318036&resultId=113321)|dotnet/runtime#78279|
|[108688](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108688)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108688&view=ms.vss-test-web.build-test-results-tab&runId=2317628&resultId=200809)|dotnet/runtime#79448|
|[108594](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108594)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108594&view=ms.vss-test-web.build-test-results-tab&runId=2314144&resultId=200809)|dotnet/runtime#79381|
|[108557](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108557)|dotnet/runtime|[ComInterfaceGenerator.Unit.Tests.WorkItemExecution](https://dev.azure.com/dnceng-public/public/_build/results?buildId=108557&view=ms.vss-test-web.build-test-results-tab&runId=2313242&resultId=200809)|dotnet/runtime#79501|
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|20|23|23|

**Comment by jakobbotsch** [comment]
@jkotas is that the right known error message? It makes it look like a duplicate of #79439, but I don't think it is.

**Comment by jakobbotsch** [comment]
The test is validating that a tailcall happens at a call site without a tail. prefix. After some recent changes we are able to inline functions that introduce an address-taken variable so we no longer tailcall.
The exact inlined path is:

```scala
  [INL01 IL=0005 TR=000007 06006F27] [INLINED: callee: below ALWAYS_INLINE size DEVIRT] System.Diagnostics.StackTrace:ToString():System.String:this
    [INL02 IL=0002 TR=000071 06006F28] [INLINED: call site: profitable inline] System.Diagnostics.StackTrace:ToString(int):System.String:this
      [INL03 IL=0005 TR=000082 06003BCE] [INLINED: callee: below ALWAYS_INLINE size] System.Text.StringBuilder:.ctor(int):this
        [INL04 IL=0007 TR=000100 06003BD2] [INLINED: call site: profitable inline] System.Text.StringBuilder:.ctor(int,int):this
          ...
          [INL12 IL=0071 TR=000124 0600055C] [INLINED: callee: aggressive inline attribute] System.GC:AllocateUninitializedArray[ushort](int,bool):ushort[]
            [INL13 IL=0003 TR=000199 06005990] [INLINED: callee: below ALWAYS_INLINE size] System.Runtime.CompilerServices.RuntimeHelpers:IsReferenceOrContainsReferences[ushort]():bool
            [INL14 IL=0041 TR=000205 06000566] [INLINED: call site: profitable inline] System.GC:<AllocateUninitializedArray>g__AllocateNewUninitializedArray|66_0[ushort](int,bool):ushort[]
              [INL15 IL=0021 TR=000231 060007C7] [INLINED: callee: below ALWAYS_INLINE size DEVIRT] System.RuntimeType:get_TypeHandle():System.RuntimeTypeHandle:this
```

where `System.Text.StringBuilder:.ctor(int,int)` is newly inlined (after #79460) and `System.GC:AllocateUninitializedArray` is the function that ends up with an address-taken local.

The variable isn't actually address exposed but we are quite conservative still (#65102 would have fixed it). Anyway, the test should not be trying to validate tailcalls without a tail prefix so I will just fix that.

**Comment by jkotas** [comment]
> @jkotas is that the right known error message?

Copy&paste ... fixed.
